sky/packages/sky/lib/src/rendering/sky_binding.dart

```diff
@@ -105,37 +105,42 @@ class SkyBinding extends HitTestTarget {
   /// A router that routes all pointer events received from the engine
   final PointerRouter pointerRouter = new PointerRouter();
 
+  /// State for all pointers which are currently down.
+  /// We do not track the state of hovering pointers because we need
+  /// to hit-test them on each movement.
   Map<int, _PointerState> _stateForPointer = new Map<int, _PointerState>();
 
-  _PointerState _createStateForPointer(sky.PointerEvent event, Point position) {
-    HitTestResult result = hitTest(position);
-    _PointerState state = new _PointerState(result: result, lastPosition: position);
-    _stateForPointer[event.pointer] = state;
-    return state;
-  }
-
-  _PointerState _getOrCreateStateForPointer(event, position) {
-    _PointerState state = _stateForPointer[event.pointer];
-    if (state == null)
-      state = _createStateForPointer(event, position);
-    return state;
-  }
-
   void _handlePointerEvent(sky.PointerEvent event) {
     Point position = new Point(event.x, event.y);
 
-    _PointerState state = _getOrCreateStateForPointer(event, position);
-
-    if (event.type == 'pointerup' || event.type == 'pointercancel') {
-      if (_hammingWeight(event.buttons) <= 1)
-        _stateForPointer.remove(event.pointer);
+    _PointerState state = _stateForPointer[event.pointer];
+    switch (event.type) {
+      case 'pointerdown':
+        if (state == null) {
+          state = new _PointerState(result: hitTest(position), lastPosition: position);
+          _stateForPointer[event.pointer] = state;
+        }
+        dispatchEvent(event, state.result);
+        break;
+      case 'pointermove':
+        if (state != null) {
+          event.dx = position.x - state.lastPosition.x;
+          event.dy = position.y - state.lastPosition.y;
+          state.lastPosition = position;
+          dispatchEvent(event, state.result);
+        } else {
+          dispatchEvent(event, hitTest(position)); // hovering
```

Comment:
None of the rest of the code is prepared to handle `pointermove` for a pointer that isn't down.  Instead, we probably should just ignore the event for now.  At some point we'll support hover, but we're explicitly ignoring it for now.